### PR TITLE
Re-Implement rename function in file and directory

### PR DESCRIPTION
**Pull request by yin19941005**
Using the new FileInformation classes and the generic encoder to implement the rename function. 

Created the FileRenameInformation class.

### CODE REVIEW

File: src/main/java/com/hierynomus/msfscc/fileinformation/FileRenameInformation.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hierynomus.msfscc.fileinformation;
+
+public class FileRenameInformation implements FileSettableInformation{
```
Comment (hierynomus) [inline src/main/java/com/hierynomus/msfscc/fileinformation/FileRenameInformation.java]: Formatting (space before curly missing) (and on some other places as well)

File: src/main/java/com/hierynomus/smbj/share/Directory.java [hunk 2]
```diff
@@ -160,10 +163,69 @@ private F prepareNext() {
...
+
+    public void rename(String newName, boolean replaceIfExist, long rootDirectory)throws TransportException, SMBApiException {
+        FileRenameInformation renameInfo = new FileRenameInformation(replaceIfExist, rootDirectory, newName);
+        this.setFileInformation(renameInfo);
+    }
+
+    /**
+     * Get information for a given fileId
+     **/
+    private <F extends FileSettableInformation> void setFileInformation(F information) throws SMBApiException, TransportException {
```
Comment (hierynomus) [inline src/main/java/com/hierynomus/smbj/share/Directory.java]: Extract to package private method on DiskEntry (the common superclass of File and Directory), also for `setInfoCommon`

**Comment by yin19941005** [comment]
Hi @hierynomus ,

I didn't got the right to check what's going wrong on Better Code Hub. Could you check what's happen?

File: src/main/java/com/hierynomus/msfscc/fileinformation/FileRenameInformation.java (hunk 1 above)
Comment (yin19941005) [inline src/main/java/com/hierynomus/msfscc/fileinformation/FileRenameInformation.java]: I cannot find the space before curly is missing on other lines, where are the others missing space?

Comment (hierynomus) [inline src/main/java/com/hierynomus/msfscc/fileinformation/FileRenameInformation.java]: They are missing before the `throws keyword` in the File/Directory classes

**Comment by hierynomus** [comment]
@yin19941005 I just deleted the BCH integration. Please do check the Codacy warnings 😄

**Comment by yin19941005** [comment]
Sorry for the late reply, I think I fixed for Codacy warnings and ready to merge now.

**Comment by hierynomus** [comment]
@yin19941005 Did you see the comment on issue #48?

**Comment by yin19941005** [comment]
Hi @hierynomus ,

I missed it previously, but I am not able reproduce the issue. Renaming the file in existing to another existing directory is working fine for my environment. You can refer to my reply on #48.

**Pull request merged**